packages/cmf/__tests__/sagaRouter/router.test.js

```diff
@@ -229,11 +268,53 @@ describe('sagaRouter route and route params', () => {
 		const gen = sagaRouter(getMockedHistory(), routes);
 		expect(gen.next().value).toEqual(take('@@router/LOCATION_CHANGE'));
 		expect(gen.next({ type: '@@router/LOCATION_CHANGE' }).value).toEqual(
-			spawn(routes['/matchingroute/:id'], { id: 'anId' })
+			spawn(routes['/matchingroute/:id'], { id: 'anId' }, true)
+		);
+		expect(gen.next(mockTask).value).toEqual(take('@@router/LOCATION_CHANGE'));
+		const expectedCancelYield = cancel(mockTask);
+		expect(gen.next({ type: '@@router/LOCATION_CHANGE' }).value).toEqual(expectedCancelYield);
+		expect(gen.next().value).toEqual(spawn(routes['/matchingroute/:id'], { id: 'anotherId' }, true));
+	});
+});
+
+describe('sagaRouter configurations', () => {
+	it('with restartOnRouteChange set to true, saga will be restarted when change to a matching route', () => {
+		const mockTask = createMockTask();
+		const routes = {
+			'/resources': {
+				restartOnRouteChange: true,
+				saga: function* resourcesSaga(params, isExact) {
+					if (isExact) {
+						yield take('SOMETHING');
+					}
+				},
+			},
+			'/resources/action': function* resourcesActionSaga(params) {},
+		};
+		function getMockedHistory() {
+			let count = 0;
+			return {
+				getCurrentLocation() {
+					if (count === 0) {
+						count = 1;
+						return {
+							pathname: '/resources',
+						};
+					}
+					return {
+						pathname: '/resources/action',
+					};
+				},
+			};
+		}
```

Comment:
Looks like that function is duplicated here https://github.com/Talend/ui/pull/1018/files#diff-aa898b17c94b680d677cafaa355f9b62R182. Could you extract it to define it only once?

Reply:
I've modify this test case, now they test on different cases.